src/iface/interface/mod.rs

```diff
@@ -720,7 +724,7 @@ pub(crate) enum IpPacket<'a> {
     #[cfg(feature = "socket-raw")]
     Raw((IpRepr, &'a [u8])),
     #[cfg(any(feature = "socket-udp", feature = "socket-dns"))]
-    Udp((IpRepr, UdpRepr, &'a [u8])),
+    Udp((Option<PacketId>, IpRepr, UdpRepr, &'a [u8])),
```

Comment:
I think it makes more sense to pass the ID alongside the `IpPacket`, not within. Any packet could get an ID, independently of whether it's UDP or not.

Reply:
Hmm, yes. I seem to have gotten some wires crossed while merging, this thing is entirely unnecessary. I've removed it.